owid/catalog/tables.py

```diff
@@ -408,10 +447,20 @@ def prune_metadata(self) -> "Table":
     def copy(self, deep: bool = True) -> "Table":
         """Copy table together with all its metadata."""
         tab = super().copy(deep=deep)
-        tab.metadata = copy.deepcopy(self.metadata)
-        tab._fields = copy.deepcopy(self._fields)
+        tab.copy_metadata_from(self)
         return tab
 
+    def copy_metadata_from(self, table: "Table") -> None:
+        """Copy metadata from a different table to self."""
+        self.metadata = dataclasses.replace(table.metadata)
+
+        # NOTE: copying with `dataclasses.replace` is much faster than `copy.deepcopy`
+        new_fields = defaultdict(VariableMeta)
+        for k, v in table._fields.items():
+            new_fields[k] = dataclasses.replace(v)
+            new_fields[k].sources = [dataclasses.replace(s) for s in v.sources]
+        self._fields = new_fields
```

Comment:
I don't fully understand what's happening here. But I'm wondering what happens if the new table does not have variables for which there is variable metadata, or, on the other hand, the new table has variables for which there is no variable metadata.
I suppose the expected behaviour could be that the function uses as much metadata as possible, but doesn't fail or warn if there is unused metadata. On the other hand, if there is missing metadata (for new variables) a warning could be raised.

Reply:
That's a good idea. I've added standard `errors: Literal["raise", "ignore", "warn"]` argument there.